x-pack/platform/plugins/shared/fleet/server/routes/data_streams/handlers.ts

```diff
@@ -257,3 +257,72 @@ export const getListHandler: RequestHandler = async (context, request, response)
     throw err;
   }
 };
+
+export const getDeprecatedILMCheckHandler: RequestHandler = async (context, request, response) => {
+  try {
+    const { elasticsearch } = await context.core;
+    const esClient = elasticsearch.client.asCurrentUser;
+
+    const DEPRECATED_ILM_POLICIES = ['logs', 'logs@lifecycle'];
+
+    // Fetch ILM policies and component templates in parallel
+    const [ilmResponse, componentTemplatesResponse] = await Promise.all([
+      esClient.ilm.getLifecycle(),
+      esClient.cluster.getComponentTemplate(),
```

Comment:
We should limit the component template check to `*@package` which are the templates managed by Fleet.

Example you can search with a wildcard like this: https://github.com/elastic/kibana/pull/243333/files#diff-76660d83c3cf706897f48ee6aab8c6ed8bb92386caf7a621fe1eaea62cd237f8

More context here: https://github.com/elastic/ingest-dev/issues/5770